app/models/hint.rb

```diff
@@ -22,7 +22,10 @@ class Hint < ApplicationRecord
 
   def self.match(searchterm)
     searchprint = fingerprint(searchterm)
-    Hint.find_by(fingerprint: searchprint)
+    hints = Hint.where(fingerprint: searchprint)
+    # Order matching hints by source; throw away nils; return first remaining
+    # Hint. If there are none, this will return nil.
+    sources.map { |source| hints.find_by(source: source) }.compact[0]
```

Comment:
I find `sources.map { |source| hints.find_by(source: source) }.compact.first` a bit more readable. That said, it's clearly clear what is happening as is so feel free to ignore :)